pkgs/applications/editors/greenfoot/default.nix

```diff
@@ -1,39 +1,62 @@
-{ lib, stdenv, fetchurl, makeWrapper, jdk }:
+{ lib, stdenv, fetchurl, openjdk, glib, wrapGAppsHook }:
 
 stdenv.mkDerivation rec {
   pname = "greenfoot";
-  version = "3.7.1";
+  version = "3.8.0";
+
   src = fetchurl {
     # We use the deb here. First instinct might be to go for the "generic" JAR
     # download, but that is actually a graphical installer that is much harder
     # to unpack than the deb.
     url = "https://www.greenfoot.org/download/files/Greenfoot-linux-${builtins.replaceStrings ["."] [""] version}.deb";
-    sha256 = "sha256-wGgKDsA/2luw+Nzs9dWb/HRHMx/0S0CFfoI53OCzxug=";
+    sha256 = "sha256-HDXmgLHS18VZVV+hCA0RgIrKRftOlV7t+fvE0pAHGjk=";
   };
 
-  nativeBuildInputs = [ makeWrapper ];
+  nativeBuildInputs = [ wrapGAppsHook ];
+  buildInputs = [ glib ];
+
+  sourceRoot = ".";
 
-  unpackPhase = ''
-    ar xf $src
-    tar xf data.tar.xz
+  preUnpack = ''
+    unpackCmdHooks+=(_tryDebData)
+    _tryDebData() {
+      if ! [[ "$1" =~ \.deb$ ]]; then return 1; fi
+      ar xf $src
+      if ! [[ -e data.tar.xz ]]; then return 1; fi
+      unpackFile data.tar.xz
+    }
```

Comment:
Could `dpkg` be used to do the unpacking. it can be tried by just adding `dpkg` to `nativeBuildInputs` because it has a setup-hook.

Reply:
Ah, that does work. Thanks, applied!

Reply:
Apply to bluej too

Reply:
Done